BootloaderCommonPkg/Library/ShellLib/CmdBoot.c

```diff
@@ -564,6 +619,10 @@ ShellCommandBootFunc (
         goto ExitBootCmd;
       }
     }
+    Status = GetExtraImageInfo(Shell, Buffer, sizeof (Buffer), &BootOption, CurrOption);
+    if (EFI_ERROR (Status)) {
+      goto ExitBootCmd;
+    }

```

Comment:
Get file name only valid when using file system. For raw partition need get LBA info. so need add this function after line 560.
Prefer update GetBootFileInfo () to add ImageIndex parameter so that it could be re-used for all images.
For PreOS image:
 if ((CurrOption->BootFlags & BOOT_FLAGS_PREOS)  != 0) {
    ShellPrint (L"PreOS flag is set, Get PreOS image:\n ");
      Status = GetBootFileInfo (Shell, Buffer, sizeof (Buffer), &BootOption, CurrOption, LoadImageTypePreOs);
      if (EFI_ERROR (Status)) {
        goto ExitBootCmd;
      }
Similar to get extra image, first get extra image count (by default 1) from user, then get each extra images based on cound.

}

Reply:
Noted and submitted patch